lib/trino-filesystem/src/test/java/io/trino/filesystem/TestLocation.java

```diff
@@ -75,10 +75,10 @@ void testParse()
         assertLocation("/foo://host:port/path", "foo://host:port/path");
 
         // special handling for file URIs without hostnames
-        assertLocation("file:/", "file", "");
-        assertLocation("file:/hello.txt", "file", "hello.txt");
-        assertLocation("file:/some/path", "file", "some/path");
-        assertLocation("file:/some@what/path", "file", "some@what/path");
+        assertLocation("file:/", "file", "/");
```

Comment:
This change is wrong. It violates the spec that "The path will not start with a slash". Note that we strip the leading slash even for raw paths.